File: app/src/main/java/com/benjaminshai/couragers/beans/Event.java
package com.benjaminshai.couragers.beans;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by bshai on 8/14/15.
 */
public class Event implements Parcelable {

    private String title;
    private String time;
    private int hasMap;
    private String mapURL;
    private int eventID;
    private String details;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getTime() {
        return time;
    }

    public void setTime(String time) {
        this.time = time;
    }

    public int getHasMap() {
        return hasMap;
    }

    public void setHasMap(int hasMap) {
        this.hasMap = hasMap;
    }

    public String getMapURL() {
        return mapURL;
    }

    public void setMapURL(String mapURL) {
        this.mapURL = mapURL;
    }

    public int getEventID() {
        return eventID;
    }

    public void setEventID(int eventID) {
        this.eventID = eventID;
    }

    public String getDetails() {
        return details;
    }

    public void setDetails(String details) {
        this.details = details;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(title);
        dest.writeString(time);
        dest.writeInt(hasMap);
        dest.writeString(mapURL);
        dest.writeInt(eventID);
        dest.writeString(details);
    }

    public static final Parcelable.Creator<Event> CREATOR = new Parcelable.Creator<Event>() {
        public Event createFromParcel(Parcel in) {
            return new Event(in);
        }

        public Event[] newArray(int size) {
            return new Event[size];
        }
    };

    public Event(Parcel in) {
        title = in.readString();
        time = in.readString();
        hasMap = in.readInt();
        mapURL = in.readString();
        eventID = in.readInt();
        details = in.readString();
    }
}

File: app/src/main/java/com/benjaminshai/couragers/beans/Event.kt
package com.benjaminshai.couragers.beans

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by bshai on 8/14/15.
 */
class Event(`in`: Parcel) : Parcelable {

    var title: String? = null
    var time: String? = null
    var hasMap: Int = 0
    var mapURL: String? = null
    var eventID: Int = 0
    var details: String? = null

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(title)
        dest.writeString(time)
        dest.writeInt(hasMap)
        dest.writeString(mapURL)
        dest.writeInt(eventID)
        dest.writeString(details)
    }

    init {
        title = `in`.readString()
        time = `in`.readString()
        hasMap = `in`.readInt()
        mapURL = `in`.readString()
        eventID = `in`.readInt()
        details = `in`.readString()
    }

    companion object CREATOR: Parcelable.Creator<Event> {
        override fun createFromParcel(`in`: Parcel): Event {
            return Event(`in`)
        }

        override fun newArray(size: Int): Array<Event?> {
            return arrayOfNulls(size)
        }
    }
}
